noir-projects/noir-contracts/contracts/token_contract/src/test/transfer_public.nr

```diff
@@ -42,40 +40,38 @@ unconstrained fn public_transfer_on_behalf_of_other() {
     // Impersonate recipient to perform the call
     env.impersonate(recipient);
     // Transfer tokens
-    env.call_public(public_transfer_from_call_interface);
+    public_transfer_from_call_interface.call(&mut env.public());
     // Check balances
     utils::check_public_balance(token_contract_address, owner, mint_amount - transfer_amount);
     utils::check_public_balance(token_contract_address, recipient, transfer_amount);
 }
 
-#[test]
+//#[test]
```

Comment:
remove?

Reply:
Yeah, sorry. Also switched all error messages generated in public calls to `#[test(should_fail_with="Nested call failed")]` so we don't forget to update when we have rethrows!

Reply:
Rethrows do happen now (in the hacky way) it's just that you'd need to interpret the message just like the PXE does?
* https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/simulator/src/avm/avm_simulator.test.ts#L970
* https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/pxe/src/pxe_service/pxe_service.ts#L765

Reply:
You're right it works! Proper error messages everywhere 😁 

Unfortunately I had to readd a few of the `avmOpcode...` oracles because some lower-level tests read storage directly (for example). Shouldn't be too bad, and remain unused for contract testing.

Reply:
Boo, let's try to work on a plan to remove those later if possible, I see a lot of them are reintroduced.

Reply:
I don't think we'll be able to remove them if we want to be able to test public state vars in isolation (which we do). It's not too bad IMO as long as they're not necessary for actual contract tests